frontend/components/Auction/index.tsx

```diff
@@ -9,48 +9,62 @@ import Image from "next/image";
 import Badge from "@/components/Badge";
 import { londrina } from "@/util/fonts";
 import classNames from "classnames";
+import { useAuctionData } from "@/hooks/useAuctionData";
+import { useEffect, useState } from "react";
 
-export default function Auction({
-  tokenId,
-  isLastToken = false,
-  endTime,
-  onEnd = () => {},
-}: {
-  tokenId: string;
-  isLastToken?: boolean;
-  endTime: number;
-  onEnd?: () => void;
-}) {
-  const { countdownString } = useCountdown(endTime, onEnd);
-  return (
+export default function Auction({}: {}) {
+  const [auctionData, currentToken, loading, incrementToken, decrementToken] =
+    useAuctionData();
+  const [endTime, setEndTime] = useState<number>(0);
+  const { countdownString, isEnded } = useCountdown(endTime, () => {});
+
+  useEffect(() => {
+    if (currentToken && currentToken.auction) {
+      setEndTime(parseInt(currentToken.auction.endTime, 10));
+    }
+  }, [currentToken]);
+
+  return !loading && currentToken ? (
     <div className="auction">
       <div className="auction--grid">
         <div className="auction--token" />
         <div className="auction--info">
           <div className="auction--row">
-            <Link href={`/${parseInt(tokenId, 10) - 1}`}>
-              <button className="auction--nav-button">{arrowLeft}</button>
-            </Link>
-            <Link
-              className={classNames({
-                disabled: isLastToken,
-              })}
-              href={`/${parseInt(tokenId, 10) + 1}`}
+            <button
+              onClick={() => {
+                decrementToken();
```

Comment:
@MrKevinOConnell I think we'd want to have this link to a new page... for example I should be able to link to `https://purple.wtf/200`, whereas the way it's structured now it'll all just appear on a single page. What do you think?

Reply:
yeah i was thinking about this - I want to say performance wise having it on the same page is better - just fetch the first x then paginate if user hits the limit. having a page for each auction seems way worse since its an API call on every click (unless if way to pass data to link. I know you can do this in react,haven't seen how to do this in next)